pinot-common/src/main/java/org/apache/pinot/common/function/FunctionRegistry.java

```diff
@@ -86,6 +96,11 @@ public static void init() {
    */
   public static void registerFunction(Method method, boolean nullableParameters) {
     registerFunction(method.getName(), method, nullableParameters);
+
+    // Calcite ScalarFunctionImpl doesn't allow customized named functions. TODO: fix me.
```

Comment:
What's the fix here?

Reply:
we will need to follow up items in https://github.com/apache/pinot/issues/8597